### PR TITLE
dev-libs/newt: bump to 0.52.20 w/ EAPI=6

### PR DESCRIPTION
**Pull request by benkohler**
Bump to upsream version 0.52.20.  Convert to EAPI=6, which requires a
small patch adjustment for -p1.  Removed now-unused eutils inherit.
Added 3.x to PYTHON_COMPAT since upstream added py3 support.  Added :0
slot on ncurses dep.

### CODE REVIEW

File: dev-libs/newt/files/newt-0.52.13-gold.patch [hunk 1]
```diff
@@ -1,5 +1,5 @@
---- configure.ac~	2011-11-09 23:50:21.000000000 +0900
-+++ configure.ac	2011-11-09 23:54:54.139482817 +0900
+--- a/configure.ac	2011-11-09 23:50:21.000000000 +0900
++++ b/configure.ac	2011-11-09 23:54:54.139482817 +0900
```
Comment (SoapGentoo) [inline dev-libs/newt/files/newt-0.52.13-gold.patch]: not important now, but if you're git/p1-izing the patches, could you also remove the dates? unnecessary bytes, add no value, and are just noisy

File: dev-libs/newt/newt-0.52.20.ebuild [hunk 2]
```diff
@@ -0,0 +1,88 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+PYTHON_COMPAT=( python2_7 python3_4 python3_5 )
```
Comment (SoapGentoo) [inline dev-libs/newt/newt-0.52.20.ebuild]: no py3.6 love? 😢

File: dev-libs/newt/newt-0.52.20.ebuild [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~mips ~ppc ~ppc64 ~sparc ~x86 ~x86-fbsd"
+IUSE="gpm tcl nls"
+
+REQUIRED_USE="${PYTHON_REQUIRED_USE}"
+
+RDEPEND="${PYTHON_DEPS}
+	>=dev-libs/popt-1.6
+	=sys-libs/slang-2*
+	elibc_uclibc? ( sys-libs/ncurses:0 )
```
Comment (SoapGentoo) [inline dev-libs/newt/newt-0.52.20.ebuild]: generally needs `:0=`

File: dev-libs/newt/newt-0.52.20.ebuild [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+	fi
+
+	sed -i Makefile.in \
+		-e 's|$(SHCFLAGS) -o|$(LDFLAGS) &|g' \
+		-e 's|-g -o|$(CFLAGS) $(LDFLAGS) -o|g' \
+		-e 's|-shared -o|$(CFLAGS) $(LDFLAGS) &|g' \
+		-e 's|instroot|DESTDIR|g' \
+		-e 's|	make |	$(MAKE) |g' \
+		-e "s|	ar |	$(tc-getAR) |g" \
+		|| die "sed Makefile.in"
```
Comment (SoapGentoo) [inline dev-libs/newt/newt-0.52.20.ebuild]: it's fine now, but could you maybe fix this properly and upstream it?

File: dev-libs/newt/newt-0.52.20.ebuild [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+	eapply_user
+	eautoreconf
+}
+
+src_configure() {
+	econf \
+		PYTHONVERS="${PYTHON}" \
+		$(use_with gpm gpm-support) \
+		$(use_with tcl) \
+		$(use_enable nls)
```
Comment (SoapGentoo) [inline dev-libs/newt/newt-0.52.20.ebuild]: in `src_compile` you're compiling against different python impls (in the same build dir!) but configuring for one (implicitly random) impl?

File: dev-libs/newt/newt-0.52.20.ebuild [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+	installit() {
+		emake \
+			DESTDIR="${D}" \
+			PYTHONVERS="${EPYTHON}" \
+			install
+		python_optimize
+	}
+	python_foreach_impl installit
+	dodoc peanuts.py popcorn.py tutorial.sgml
+	doman whiptail.1
```
Comment (SoapGentoo) [inline dev-libs/newt/newt-0.52.20.ebuild]: if it has some standard crap (`README*`, `ChangeLog` etc), don't forget to call `einstalldocs`

Comment (benkohler) [reply dev-libs/newt/newt-0.52.20.ebuild]: Fixed

File: dev-libs/newt/newt-0.52.20.ebuild [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+inherit python-r1 multilib autotools toolchain-funcs
+
+DESCRIPTION="Redhat's Newt windowing toolkit development files"
+HOMEPAGE="https://fedorahosted.org/newt/"
+SRC_URI="https://fedorahosted.org/releases/n/e/newt/${P}.tar.gz"
+
+LICENSE="LGPL-2"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~mips ~ppc ~ppc64 ~sparc ~x86 ~x86-fbsd"
+IUSE="gpm tcl nls"
```
Comment (SoapGentoo) [inline dev-libs/newt/newt-0.52.20.ebuild]: sort alphabetically

File: dev-libs/newt/newt-0.52.20.ebuild [hunk 8]
```diff
@@ -0,0 +1,92 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+PYTHON_COMPAT=( python2_7 python3_4 python3_5 python3_6 )
```
Comment (SoapGentoo) [inline dev-libs/newt/newt-0.52.20.ebuild]: compress this down to `PYTHON_COMPAT=( python2_7 python3_{4,5,6} )`

**Comment by monsieurp** [comment]
@benkohler ping?

**Comment by benkohler** [comment]
Is there anything else that I need to do here, besides report the Makefile.in stuff upstream?

**Pull request closed without merging**